Extract default value of strict mode as config

andrewlee302 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #115322

Summary:
Strict mode is currently an argument of export API (several places)
and the default value is True, which is explicitly indicated in
the API signature. Shortcomings:
1. Maintenance issue: default values are scattered in external/
   internal places.
2. Dev test issue: we have to replicate the test cases for non-strict
   mode from the strict mode tests, which introduces code redundancy and
   test case divergence.

To address them, config is a handy choice. It's also the choice that dynamo
choose and the test dev is easier to be supported by switching to a different 
config treatment.

andrewlee302 committed:
Extract default value of strict mode as config

Summary:
Strict mode is currently an argument of export API (serveral places)
and the default value is True, which is explicitly indicated in
the API signature. Shortcomings:
1. Maintenance issue: default values are scattered in external/
   internal places.
2. Dev test issue: we have to replicate the test cases for non-strict
   mode from the strict mode tests, which introduces code redundance and
   test case divergence.
To address it, config is a handy choice. It's also the choice that
dynamo choose and the test dev is easier to be supported by swtiching to
a different config treatment.

[ghstack-poisoned]

zhxchen17 reviewed (changes requested):
Since we want this only for unittest, I think it's a bit too much to make this a configurable. Have we considered using mock here? https://docs.python.org/3/library/unittest.mock.html

andrewlee302 committed:
Update on "Extract default value of strict mode as config"


Summary:
Strict mode is currently an argument of export API (several places)
and the default value is True, which is explicitly indicated in
the API signature. Shortcomings:
1. Maintenance issue: default values are scattered in external/
   internal places.
2. Dev test issue: we have to replicate the test cases for non-strict
   mode from the strict mode tests, which introduces code redundancy and
   test case divergence.

To address them, config is a handy choice. It's also the choice that dynamo
choose and the test dev is easier to be supported by switching to a different 
config treatment.

[ghstack-poisoned]
